catalog/resource_lakehouse_monitor.go

```diff
@@ -0,0 +1,120 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const DefaultProvisionTimeout = 15 * time.Minute
+
+func WaitForMonitor(w *databricks.WorkspaceClient, ctx context.Context, monitorName string) error {
+	return retry.RetryContext(ctx, DefaultProvisionTimeout, func() *retry.RetryError {
+		endpoint, err := w.LakehouseMonitors.GetByFullName(ctx, monitorName)
+		if err != nil {
+			return retry.NonRetryableError(err)
+		}
+
+		if endpoint.Status == catalog.MonitorInfoStatusMonitorStatusActive {
+			return nil
+		}
+		if endpoint.Status == catalog.MonitorInfoStatusMonitorStatusError || endpoint.Status == catalog.MonitorInfoStatusMonitorStatusFailed {
+			return retry.NonRetryableError(fmt.Errorf("monitor status retrund %s for monitor: %s", endpoint.Status, monitorName))
+		}
+		return retry.RetryableError(fmt.Errorf("monitor %s is still pending", monitorName))
+	})
+}
+
+func ResourceLakehouseMonitor() common.Resource {
+	monitorSchema := common.StructToSchema(
+		catalog.MonitorInfo{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			common.CustomizeSchemaPath(m, "assets_dir").SetRequired()
+			common.CustomizeSchemaPath(m, "output_schema_name").SetRequired()
```

Comment:
These can be removed, based on the specification: these fields are not marked `omitempty`, so they will be required in the Terraform schema.

Reply:
The struct we are using here is the `MonitorInfo` and not `CreateMonitor`. `MonitorInfo` has additional fields such as `drift_metrics_table_name`, `profile_metrics_table_name` etc that are not present in `CreateSchema`. Therefore if we use `CreateMonitor`, the GET request is not translating all the fields correctly. 

In the `MonitorInfo` struct, assets_dir, output_schema_name and table_name are not set to `omitempty`. So this forces them to be required.

Reply:
Thanks for clarifying. This is fine then.